backend/controllers/calculation_controller.go

```diff
@@ -0,0 +1,143 @@
+package controllers
+
+import (
+	"fmt"
+	"mmr/backend/mmr"
+	view "mmr/backend/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/intinig/go-openskill/rating"
+	"github.com/intinig/go-openskill/types"
+)
+
+type CalculationController struct{}
+
+// SubmitMMRCalculation godoc
+//
+//	@Summary		Submit an MMR calculation request
+//	@Description	Submit two teams' details for MMR calculation
+//	@Tags 			MMR
+//	@Accept			json
+//	@Produce		json
+//	@Param			request	body		view.MMRCalculationRequest	true	"MMR Calculation Request"
+//	@Success		200		{object}	view.MMRCalculationResponse	"MMR calculation result"
+//	@Router			/v2/mmr/calculate [post]
+func (m CalculationController) SubmitMMRCalculation(c *gin.Context) {
+	var req view.MMRCalculationRequest
+	err := c.ShouldBindJSON(&req)
+
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	
+	ensurePlayers(c, req)
+
+	// Create players for Team 1
+	player1 := m.createPlayer(req.Team1.Players[0])
+	player2 := m.createPlayer(req.Team1.Players[1])
+
+	team1 := mmr.TeamV2{
+		Players: []mmr.PlayerV2{player1, player2},
+		Score:   int16(req.Team1.Score),
+	}
+
+	// Create players for Team 2
+	player3 := m.createPlayer(req.Team2.Players[0])
+	player4 := m.createPlayer(req.Team2.Players[1])
+
+	team2 := mmr.TeamV2{
+		Players: []mmr.PlayerV2{player3, player4},
+		Score:   int16(req.Team2.Score),
+	}
+
+	// Calculate new MMR
+	team1, team2 = mmr.CalculateNewMMRV2(&team1, &team2)
+
+	// Prepare response
+	response := view.MMRCalculationResponse{
+		Team1: m.createTeamResult(req.Team1.Score, team1),
+		Team2: m.createTeamResult(req.Team2.Score, team2),
+	}
+
+	// Respond with the updated team data
+	c.JSON(http.StatusOK, response)
+}
+
+// Checks if the player IDs from both teams are unique and that there are exactly 4 unique players.
+// If any validation fails, it responds with an appropriate error message and aborts the request.
+func ensurePlayers(c *gin.Context, req view.MMRCalculationRequest) {
+	// Extract all player IDs into a single slice
+	var playerIDs []int64
+
+	// Add all player IDs from Team 1
+	for _, player := range req.Team1.Players {
```

Comment:
Why not use the map here instead of creating a temporary array that we barely use?